btcclient/client_block_subscriber.go

```diff
@@ -51,15 +51,6 @@ func NewWithBlockSubscriber(cfg *config.BTCConfig, retrySleepTime, maxRetrySleep
 			return nil, err
 		}
 
-		// ensure we are using bitcoind as Bitcoin node, as zmq is only supported by bitcoind
-		backend, err := rpcClient.BackendVersion()
-		if err != nil {
-			return nil, fmt.Errorf("failed to get BTC backend: %v", err)
-		}
-		if backend != rpcclient.BitcoindPost25 {
-			return nil, fmt.Errorf("zmq is only supported by bitcoind, but got %v", backend)
-		}
-
```

Comment:
why remove this check?

Reply:
Might be wrong, but I don't think this check is necessary, we already check if we are using bitcoind in switch statement, and also bitcoind versions pre 25 also support zmq

Reply:
got it, the removal of dump private key will not be done in this PR, right? 
https://github.com/babylonlabs-io/vigilante/pull/2#issuecomment-2271712255

Reply:
No in this on, will be a separate PR